nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -212,6 +212,86 @@ with lib;
       description = "Path to root SSL certificate for stapling and client certificates.";
     };
 
+    # Since this is just an add_header, in principle it could be configured at
+    # the location level, but in practice I can't imagine that often being what
+    # people want, and they can always use `extraConfig` if necessary.
+    strictTransportSecurity = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable HTTP Strict Transport Security (HSTS). This sends
+          the header "strict-transport-security: max-age=63072000" on all
+          responses, which tells clients to refuse to use an insecure HTTP
+          connection for this host for the next 63072000 seconds (approximately
+          2 years), or for the duration specified by
+          `strictTransportSecurity.seconds` if set. This helps protects against
+          man-in-the-middle attacks that e.g. block HTTPS connections in the
+          hope that the client will fall back to an insecure HTTP connection,
+          which can be intercepted and modified.
+
+          The strict-transport-security header is (sent but) ignored on HTTP
+          connections, so this setting will only take effect if your website
+          actually is available over HTTPS.
+
+          HSTS only protects clients who have managed to connect at least once
+          without the man in the middle in the last two years: see also
+          `strictTransportSecurity.preload` for how to protect even the very
+          first connection.
+
+          HSTS is recommended in cases where you can reliably serve HTTPS, but
+          very annoying if you temporarily set it up but then want to use HTTP
+          again, so it is not enabled by default.
+        '';
+      };
+
+      seconds = mkOption {
+        type = types.int;
```

Comment:
```suggestion
        type = types.ints.positive;
```

Reply:
I think 0 is also an allowed value here, and makes some sense to ask clients to clear HSTS settings, so `types.ints.unsigned`?

Reply:
(I switched to `unsigned`, but lmk if you disagree)